src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7188,6 +7188,13 @@ private int GetReceiverSlotForMemberPostConditions(MethodSymbol? method)
                 }
             }
 
+            if (current.ContainingSymbol is FieldSymbol)
+            {
+                // Functions inside a field initializer are effectively static.
```

Comment:
>// Functions inside a field initializer are effectively static. [](http://example.com/codeflow?start=16&length=63)

Is this accurate? For example, it looks like in the following scenario both the lambda and the local function are implemented by an instance method.
```
#nullable enable

using System;
using System.Diagnostics.CodeAnalysis;

public class C(string p)
{
    public static string? field;

    public Action field2 = () =>
    {
        init();
        Console.WriteLine(field.Length);

        [MemberNotNull(nameof(field))]
        void init() => field ??= p;
    };
    
    string M() => p;
}
```

Reply:
You're right, when marked `static` the functions cannot use `p`. I'll adjust the comment and add the test.

Reply:
It looks like the comment got removed, but it feels like we still need some rational to explain why returning 0 is probably not 100% accurate, but OK thing to do.